src/modules/workspace/hooks/useIsWorkspaceReady.ts

```diff
@@ -47,7 +49,9 @@ export const useIsWorkspaceReady = ({
     isVaultTransactionsFetching,
   });
 
-  if (isSignInpage || (isFilteringInProgress && !isFromDapp)) {
+  const expired = searchParams.get('expired') === 'true';
+
+  if ((isSignInpage && !expired) || (isFilteringInProgress && !isFromDapp)) {
```

Comment:
add consts with description name of condition, for example:

`const onLogout = isSignInpage && !expired`
`const onFilter = isFilteringInProgress && !isFromDapp`

and:

`if(onLogout)     return { isWorkspaceReady: true, isFilteringInProgress };`

`if(onFilter)    return { isWorkspaceReady: true, isFilteringInProgress };`

Reply:
Done! I've also moved the `expired` param check to `useQueryParams` hook to keep the pattern.